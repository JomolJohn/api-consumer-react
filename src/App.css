.container{
    margin: 30px;
}
.container h4{
    text-align: center;
}