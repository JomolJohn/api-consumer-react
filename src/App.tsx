import {
  BrowserRouter as Router,
  Routes,
  Route,
} from "react-router-dom";
import Header from './components/Header';
import Home from './components/Home';
import Book from './components/book/Book';
import NotFound from './components/NotFound';
import './App.css';

function App() {
  return (
    <Router>
      <Header />
      <div className='container'>
        <Routes>
          <Route path='/' element={< Home />}></Route>
          <Route path='*' element={< NotFound />}></Route>
          <Route path="/book/:bookId" element={< Book />}></Route>
        </Routes>
      </div>
    </Router>
  );
}

export default App;
