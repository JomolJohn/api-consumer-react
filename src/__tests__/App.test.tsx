import { shallow } from 'enzyme';
import App from '../App';

describe(" App Component", () => {
    it("should render app component", () => {
      const component = shallow(<App />);
      expect(component.getElements()).toMatchSnapshot();
    });
});