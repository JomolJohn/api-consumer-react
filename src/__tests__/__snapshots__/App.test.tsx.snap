// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` App Component should render app component 1`] = `
Array [
  <BrowserRouter>
    <Header />
    <div
      className="container"
    >
      <Routes>
        <Route
          element={<Home />}
          path="/"
        />
        <Route
          element={<NotFound />}
          path="*"
        />
        <Route
          element={<Book />}
          path="/book/:bookId"
        />
      </Routes>
    </div>
  </BrowserRouter>,
]
`;
